### DISCUSSION TITLE
should emit status updates for reports?

### DISCUSSION THREAD

**Discussion by ericblade** [ericblade/mws-advanced#183] [General]
I'm not sure if this goes against any established patterns or not, but according to my TODO right above "requestAndDownloadReport()" I was considering the idea of making the MWS object also an Emitter, so that one could easily pick up status updates from functions that may take a very long time to complete asynchronously.

**Comment by pwrnrd** [comment]
I would like that, because that would allow me to create my own log entries instead of the hard coded console.logs/warns.

**Comment by ericblade** [comment]
I'm not sure if there's a more appropriate pattern that would be better suited for modern code style or not, it seems that the world has been moving away from Event Emitters . . . something to do some research and thinking about.
